File: src/MultipleEntrySeparator.cs
using System;

namespace Clipman
{
    internal static class MultipleEntrySeparator
    {
        public static string NormalizeMode(string value)
        {
            switch ((value ?? string.Empty).Trim().ToUpperInvariant())
            {
                case "NONE": return "None";
                case "NEWLINE": return "NewLine";
                case "SPACE": return "Space";
                case "COMMASPACE": return "CommaSpace";
                case "CUSTOM": return "Custom";
                case "BLANKLINE":
                default: return "BlankLine";
            }
        }

        public static string Resolve(string mode, string custom)
        {
            switch (NormalizeMode(mode))
            {
                case "None": return string.Empty;
                case "NewLine": return Environment.NewLine;
                case "Space": return " ";
                case "CommaSpace": return ", ";
                case "Custom": return DecodeCustom(custom);
                default: return Environment.NewLine + Environment.NewLine;
            }
        }

        private static string DecodeCustom(string value)
        {
            return (value ?? string.Empty)
                .Replace("\\r\\n", "\r\n")
                .Replace("\\n", "\n")
                .Replace("\\r", "\r")
                .Replace("\\t", "\t");
        }
    }
}

File: ClipmanMac/Sources/ClipmanCore/MultipleEntrySeparator.swift
import Foundation

public enum MultipleEntrySeparator {
    public static func normalize(_ value: String?) -> String {
        switch (value ?? "").trimmingCharacters(in: .whitespacesAndNewlines).lowercased() {
        case "none": return "None"
        case "newline": return "NewLine"
        case "space": return "Space"
        case "commaspace": return "CommaSpace"
        case "custom": return "Custom"
        default: return "BlankLine"
        }
    }

    public static func resolve(mode: String, custom: String) -> String {
        switch normalize(mode) {
        case "None": return ""
        case "NewLine": return "\n"
        case "Space": return " "
        case "CommaSpace": return ", "
        case "Custom":
            return custom
                .replacingOccurrences(of: "\\r\\n", with: "\r\n")
                .replacingOccurrences(of: "\\n", with: "\n")
                .replacingOccurrences(of: "\\r", with: "\r")
                .replacingOccurrences(of: "\\t", with: "\t")
        default: return "\n\n"
        }
    }
}
